### ISSUE TITLE
Chrome crashes when choosing large files with the sample project

### ISSUE THREAD

**Issue by bzuillsmith** [gdsestimating/three-dxf#5]
I tried uploading a 30 MB file into the sample project and Chrome crashed while loading it. Haven't tried other browsers yet.

**Comment by openhardwarecoza** [comment]
What was the job?  A raster,  dxf,  svg or gcode? 
Can you submit their  file for testing?

**Comment by bzuillsmith** [comment]
This is specific to the three-dxf library and is in the three-dxf issues, not LaserWeb.

**Comment by openhardwarecoza** [comment]
oh! Sorry! Was working on my phone earlier and didn't notice! My bad!

**Comment by openhardwarecoza** [comment]
For what it's worth, we had this issue on LaserWeb when loading large files, which is why i felt surprised when i thought it returned.  

Sven, did a great fix for us:  
https://github.com/openhardwarecoza/LaserWeb/commit/73d98605a790a6a2d8bcfdd204c186fbc4c8ccb6

And he explained it to me better in the comments on https://github.com/openhardwarecoza/LaserWeb/commit/5acf749434ea98fc9f047246fabcae2bb8f0bb62

Which I implemented for Laserweb's gcode viewer here https://github.com/openhardwarecoza/LaserWeb/commit/41ee83ed3e7124e4cf346e2cf78de991648e818
Basically, it runs chunks of crunching, thus not tying up the thread and having chrome die (:

Might be of use here too, just need to fighure out which for loop is the longest running one

**Comment by bzuillsmith** [comment]
Closing this because it's a known issue and there isn't a big push to solve it at the moment. This would probably best be handled by moving the work to a web worker.
